### PR TITLE
ELEMENTS-1212: add alwaysVisible property to table data column

### PR DESCRIPTION
**Pull request by andreacornaglia**

### CODE REVIEW

File: ui/nuxeo-data-table/iron-data-table.js [hunk 1]
```diff
@@ -809,7 +809,10 @@ import '../nuxeo-button-styles.js';
       tableSettings.columns = {};
       if (this.columns) {
         this.columns.forEach((column, idx) => {
-          tableSettings.columns[column.field ? column.field : `col-${idx}`] = { hidden: column.hidden };
+          tableSettings.columns[column.field ? column.field : `col-${idx}`] = {
+            hidden: column.hidden,
+            alwaysVisible: column.alwaysVisible,
```
Comment (Gabez0r) [inline ui/nuxeo-data-table/iron-data-table.js:814]: I was not expecting us to need this, since we cannot change `alwaysVisible` from the table settings. What is the scenario where we need this?

Comment (andreacornaglia) [reply ui/nuxeo-data-table/iron-data-table.js:814]: @Gabez0r I thought we needed this for the data-table-settings component - will double-check.

File: ui/nuxeo-data-table/data-table-settings.js [hunk 2]
```diff
@@ -56,7 +56,7 @@ import '../nuxeo-button-styles.js';
                 <div class="row layout horizontal">
                   <dom-repeat items="[[columns]]" as="column">
                     <template>
-                      <tr>
+                      <tr hidden="{{column.alwaysVisible}}">
```
[the remark this answers is not in the document]
Comment (andreacornaglia) [reply ui/nuxeo-data-table/data-table-settings.js:59]: cool, I'll check it out :)

File: ui/nuxeo-data-table/data-table-column.js [hunk 3]
```diff
@@ -86,6 +86,14 @@ import './data-table-column-filter.js';
           value: false,
         },
 
+        /**
+         * If `true`, the column will not be shown in the settings dialog
```
[the remark this answers is not in the document]
Comment (andreacornaglia) [reply ui/nuxeo-data-table/data-table-column.js]: yes! done

**Comment by semisse** [comment]
Just a personal note: I finding it a little bit confusing using `alwaysVisible` and, specially, `notAlwaysVisible` nomenclature. Is `neverVisible` better? Probably not but maybe this can spark some better alternatives.

**Comment by andreacornaglia** [comment]
> Just a personal note: I finding it a little bit confusing using `alwaysVisible` and, specially, `notAlwaysVisible` nomenclature. Is `neverVisible` better? Probably not but maybe this can spark some better alternatives.

@semisse yeah I know the name is not great - the idea is to say alwaysVisible in the sense that the column will always appear on the table and cannot be hidden from the dialog settings. I thought of using just fixed or visible but that looks too similar to css properties so I wanted to give it a specific name.

File: ui/nuxeo-data-table/data-table-settings.js [hunk 4]
```diff
@@ -131,6 +131,10 @@ import '../nuxeo-button-styles.js';
         this._resetSettings();
       }
     }
+
+    isNotAlwaysVisible(column) {
```
Comment (nmpcunha) [inline ui/nuxeo-data-table/data-table-settings.js:135]:
```suggestion
    canChangeVisibility(column) {
```

File: ui/nuxeo-data-table/iron-data-table.js [hunk 5]
```diff
@@ -809,7 +809,9 @@ import '../nuxeo-button-styles.js';
       tableSettings.columns = {};
       if (this.columns) {
         this.columns.forEach((column, idx) => {
-          tableSettings.columns[column.field ? column.field : `col-${idx}`] = { hidden: column.hidden };
+          tableSettings.columns[column.field ? column.field : `col-${idx}`] = {
+            hidden: column.hidden,
+          };
```
Comment (nmpcunha) [inline ui/nuxeo-data-table/iron-data-table.js:814]: This kind of change can be on a separate commit for formatting and/or cleanup. It helps the reviewers. 😉

File: ui/nuxeo-data-table/data-table-settings.js (hunk 4 above)
Comment (semisse) [reply ui/nuxeo-data-table/data-table-settings.js:135]: This is more clear to me

File: ui/nuxeo-data-table/iron-data-table.js (hunk 5 above)
Comment (Gabez0r) [reply ui/nuxeo-data-table/iron-data-table.js:814]: I agree. As a matter o fact, unless this was fixed by our linting tools, I guess we don't even need to touch this.

File: ui/nuxeo-data-table/data-table-settings.js (hunk 4 above)
Comment (andreacornaglia) [reply ui/nuxeo-data-table/data-table-settings.js]: thanks! changed!

File: ui/nuxeo-data-table/iron-data-table.js (hunk 5 above)
Comment (andreacornaglia) [reply ui/nuxeo-data-table/iron-data-table.js]: thanks! reverted that change!

**Comment by nmpcunha** [comment]
I've forced a pipeline run after the issue with `packages.nuxeo.com` was solved. (cc @andreacornaglia )
